packages/web3/src/utils/error.ts

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+class CompilationError {
+  constructor(
+    public lineStart: number,
+    public column: number,
+    public errorType: string,
+    public line: number,
+    public codeLine: string,
+    public errorIndicator: string,
+    public message: string,
+    public additionalLine1?: string,
+    public additionalLine2?: string
+  ) {}
+
+  reformat(line: number, file: string): string {
+    const spaces = `${line}`.replace(/\d/g, ' ')
+    const newError = `${file} (${line}:${this.column}): ${this.errorType}
+${line} |${this.codeLine}
+${spaces} |${this.errorIndicator}
+${spaces} |${this.message}`
+
+    if (this.additionalLine1 && this.additionalLine2) {
+      return `${newError}\n${spaces} |${this.additionalLine1}\n${spaces} |${this.additionalLine2}`
+    } else {
+      return newError
+    }
+  }
+}
+
+const errorRegex = /error \((\d+):(\d+)\):\s*(.*)\n\s*(\d+)\s*\|(.*)\n.*\|(.*)\n\s*\|(.*)(?:\n\s*\|(.*)\n\s*\|(.*))?/
+
+export function parseError(error: string): CompilationError | undefined {
+  const match = error.match(errorRegex)
```

Comment:
[Nit] If for some reason this doesn't match the error regex, does it make sense to fall back to the original error? Otherwise I guess we might get regex error below.

Reply:
we throw it [here](https://github.com/alephium/alephium-web3/blob/1a9019ae2678ba60e9a81fdf49954966831e7892/packages/web3/src/contract/contract.ts#L477-L480) when we use the parsing function.